lib/std/c.zig

```diff
@@ -408,11 +408,17 @@ pub extern "c" fn setlogmask(maskpri: c_int) c_int;
 
 pub extern "c" fn if_nametoindex([*:0]const u8) c_int;
 
-pub usingnamespace if (builtin.os.tag == .linux and builtin.target.isMusl()) struct {
-    // musl does not implement getcontext
-    pub const getcontext = std.os.linux.getcontext;
-} else struct {
-    pub extern "c" fn getcontext(ucp: *std.os.ucontext_t) c_int;
+pub usingnamespace switch (builtin.os.tag) {
+    .linux => if (builtin.target.isMusl()) struct {
+        // musl does not implement getcontext
+        pub const getcontext = std.os.linux.getcontext;
+    } else struct {
+        pub extern "c" fn getcontext(ucp: *std.os.ucontext_t) c_int;
+    },
+    .openbsd => struct {},
+    else => struct {
+        pub extern "c" fn getcontext(ucp: *std.os.ucontext_t) c_int;
+    },
 };
```

Comment:
```suggestion
pub extern "c" fn getcontext(ucp: *std.os.ucontext_t) c_int;
```

 * don't fear linker errors
 * std.c should not re-export something from std.os.linux
 * avoid `usingnamespace`